dash/_callback.py

```diff
@@ -467,27 +467,28 @@ def add_context(*args, **kwargs):
             if NoUpdate.is_no_update(output_value):
                 raise PreventUpdate
 
-            if no_output:
+            component_ids = collections.defaultdict(dict)
+
+            if not has_output:
```

Comment:
nit: you could reverse the `if` to make it `if has_output:` since there's an else after.